homeassistant/components/alarmdecoder.py

```diff
@@ -99,13 +98,10 @@ def async_setup(hass, config):
     path = DEFAULT_DEVICE_PATH
     baud = DEFAULT_DEVICE_BAUD
 
-    sync_connect = asyncio.Future(loop=hass.loop)
-
     def handle_open(device):
         """Handle the successful connection."""
         _LOGGER.info("Established a connection with the alarmdecoder")
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_alarmdecoder)
-        sync_connect.set_result(True)
+        run_callback_threadsafe(hass.loop, handle_connect)
```

Comment:
I'm not sure that you need this at all (see my comment at the end), but if it should be kept you should use `hass.add_job` instead to schedule the callback for the event loop.